tests/test_plot.py

```diff
@@ -17,36 +17,38 @@
 from rasterio.enums import ColorInterp
 
 
-@pytest.mark.skipif(plt is None,
-                    reason="requires matplotlib")
-def test_show_raster():
-    """
-    This test only verifies that code up to the point of plotting with
-    matplotlib works correctly.  Tests do not exercise matplotlib.
-    """
+@pytest.importorskip('matplotlib')
+def test_show_raster_band():
+    """Test plotting a single raster band."""
     with rasterio.open('tests/data/RGB.byte.tif') as src:
-        try:
-            show((src, 1))
-            fig = plt.gcf()
-            plt.close(fig)
-        except ImportError:
-            pass
+        show((src, 1))
+        fig = plt.gcf()
+        plt.close(fig)
 
-        with rasterio.open('tests/data/RGB.byte.tif') as src:
-            try:
-                show(src)
-                fig = plt.gcf()
-                plt.close(fig)
-            except ImportError:
-                pass
+@pytest.importorskip('matplotlib')
+def test_show_raster_mult_bands():
+    """Test multiple bands plotting."""
```

Comment:
`importorskip` isn't a decorator. Please delete line 28 and add

`matplotlib = pytest.importorskip('matplotlib')`

after line 30. The same change will need to be made at other places in this file.

Reply:
ok cool - latest push implements this, let's see if that works :-)